crates/p2p/src/swarm/errors.rs

```diff
@@ -66,7 +66,27 @@ pub enum SetupUpgradeError {
     UnexpectedStreamClose,
 }
 
-/// Errors from libp2p
+/// Generic errors that can occur during message operations.
+#[derive(Debug, Error)]
+pub enum MessageError {
```

Comment:
Hmm.. Didn't we have such errors before this PR? Let's name it `GossipError` or something.